error removing FK

```
2016-01-19 17:20:19,946 [ERROR] metabase.middleware :: DELETE /api/foreignkey/16 500 (28 ms)
{:message
 "No implementation of can-write? for class metabase.models.foreign_key.ForeignKeyInstance; please provide one.",
 :stacktrace
 ["metabase.models.interface$fn__8367.doInvoke(interface.clj:193)"
  "metabase.models.interface$eval8148$fn__8177$G__8127__8186.invoke(interface.clj:41)"
  "metabase.api.common$write_check.invoke(common.clj:442)"
  "metabase.api.foreignkey$fn__26539$fn__26540.invoke(foreignkey.clj:12)"
  "metabase.api.common.internal$wrap_catch_api_exceptions.invoke(internal.clj:216)"
  "metabase.api.foreignkey$fn__26539.invoke(foreignkey.clj:9)"
  "metabase.middleware$enforce_authentication$fn__18212.invoke(middleware.clj:83)"
  "metabase.api.routes$fn__28436.invoke(routes.clj:43)"
  "metabase.routes$fn__28479.invoke(routes.clj:23)"
  "metabase.middleware$log_api_call$fn__18333.invoke(middleware.clj:284)"
  "metabase.middleware$add_security_headers$fn__18265.invoke(middleware.clj:192)"
  "metabase.middleware$bind_current_user$fn__18216.invoke(middleware.clj:96)"
  "metabase.middleware$wrap_current_user_id$fn__18190.invoke(middleware.clj:65)"
  "metabase.middleware$wrap_api_key$fn__18231.invoke(middleware.clj:107)"
  "metabase.middleware$wrap_session_id$fn__18183.invoke(middleware.clj:56)"]}
```

--- a/src/metabase/api/foreignkey.clj
+++ b/src/metabase/api/foreignkey.clj
@@ -2,7 +2,7 @@
   "/api/foreignkey endpoints."
   (:require [compojure.core :refer [DELETE]]
             [metabase.api.common :refer :all]
-            [metabase.db :refer :all]
+            [metabase.db :as db]
             (metabase.models [foreign-key :refer [ForeignKey]])
             [metabase.driver :as driver]))
 
@@ -10,6 +10,7 @@
   "Delete a `ForeignKey`."
   [id]
   (write-check ForeignKey id)
-  (del ForeignKey :id id))
+  (db/cascade-delete ForeignKey :id id)
+  {:success true})
 
 (define-routes)

--- a/src/metabase/models/foreign_key.clj
+++ b/src/metabase/models/foreign_key.clj
@@ -14,7 +14,9 @@
   i/IEntity
   (merge i/IEntityDefaults
          {:types        (constantly {:relationship :keyword})
-          :timestamped? (constantly true)}))
+          :timestamped? (constantly true)
+          :can-read?    (constantly true)
+          :can-write?   i/superuser?}))
 
 
 (u/require-dox-in-this-namespace)

--- a/test/metabase/api/foreignkey_test.clj
+++ b/test/metabase/api/foreignkey_test.clj
@@ -1,0 +1,51 @@
+(ns metabase.api.foreignkey-test
+  "tests for /api/foreignkey api endpoints"
+  (:require [clojure.tools.macro :refer [symbol-macrolet]]
+            [expectations :refer :all]
+            [metabase.db :as db]
+            (metabase [http-client :as http]
+                      [middleware :as middleware])
+            (metabase.models [database :refer [Database]]
+                             [field :refer [Field]]
+                             [foreign-key :refer :all]
+                             [table :refer [Table]])
+            [metabase.test.util :as tu]
+            [metabase.test.data.users :refer :all]))
+
+
+;; ## /api/segment/* AUTHENTICATION Tests
+;; We assume that all endpoints for a given context are enforced by the same middleware, so we don't run the same
+;; authentication test on every single individual endpoint
+
+(expect (get middleware/response-unauthentic :body) (http/client :delete 401 "foreignkey/123"))
+
+
+;; ## DELETE /api/foreignkey/:id
+
+;; test security.  requires superuser perms
+(expect "You don't have permissions to do that."
+  ((user->client :rasta) :delete 403 "foreignkey/1"))
+
+
+(expect
+  [{:success true}
+   nil]
+  (tu/with-temp Database [{database-id :id} {:name      "FK Test"
+                                             :engine    :yeehaw
+                                             :details   {}
+                                             :is_sample false}]
+    (tu/with-temp Table [{table-id :id} {:name   "FK Test"
+                                         :db_id  database-id
+                                         :active true}]
+      (tu/with-temp Field [{field-id :id} {:table_id    table-id
+                                           :name        "FK Test"
+                                           :base_type   :TextField
+                                           :field_type  :info
+                                           :active      true
+                                           :preview_display true
+                                           :position    1}]
+        (tu/with-temp ForeignKey [{:keys [id]} {:destination_id field-id
+                                                :origin_id      field-id
+                                                :relationship   "whoot"}]
+          [((user->client :crowberto) :delete 200 (format "foreignkey/%d" id))
+           (db/sel :one ForeignKey :id id)])))))

